internal/provisioner/model/names.go

```diff
@@ -79,6 +79,23 @@ func (c *Contour) EnvoyRBACNames() RBACNames {
 	}
 }
 
+// ContourDeploymentLabels returns labels for a Contour deployment.
+func (c *Contour) ContourDeploymentLabels() map[string]string {
```

Comment:
I suggest a minor rename to avoid overloading the term "Deployment":

```suggestion
// ComponentLabels returns labels for a Contour component.
func (c *Contour) ComponentLabels() map[string]string {
```